### PR TITLE
Fix can't remove data in alluxio when under fs is readonly

### PR DESCRIPTION
**Pull request by bzheng888**
Fix can't remove data in alluxio when under fs is readonly

### What changes are proposed in this pull request?
I think this is a problem. When underfs is readonly, it should not restrict us to remove data from alluxio.

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

**Review by maobaolong — APPROVED** [review]
Good catch LGTM

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1796]: @yuzhu Do you think we should allow rm alluxioOnly files when a mount point is read-only?

Comment (bzheng888) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1796]: @jiacheliu3 We have a scenario where the underfs is read-only and the underfs's data changes. When I execute ./alluxio fs checkConsistency -r /, there are some inconsistent files that should be deleted cannot be deleted.

Comment (maobaolong) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1796]: Another case, ./alluxio fs rm --alluxioOnly /test , the file cannot  be removed too.

Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1796]: But what if the UFS file does not change and you just delete the file in Alluxio? The change will not propagate the UFS and will not be rejected. Later the file will appear again due to sync. This will be confusing.

Comment (maobaolong) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1796]: @jiacheliu3 Yeah, the purpose of deleting the file is that the file was never be consistent with the ufs, so we want to delete it from Alluxio, and then, the new consistent inode will be created.

**Comment by maobaolong** [comment]
Hi @jiacheliu3 Do you have any thoughts?

**Comment by HelloHorizon** [comment]
@jiacheliu3 can you take another look?

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java
Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1796]: Thanks for the explanation, makes sense to me

Comment (jiacheliu3) [reply core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1796]: I'm actually a little concerned about files that are not in UFS. For example you `--alluxioOnly`  remove an ASYNC_THROUGH file before it's persisted, or remove a MUST_CACHE file. These can all lead to data loss.

What about we only allow `--alluxioOnly` remove a file only if it's persisted? @yuzhu @maobaolong

**Review by jiacheliu3 — CHANGES_REQUESTED** [review]
Thanks for the contribution! Left some comments, more to start a quick discussion.

Also could you add a unit test? Thx!

**Comment by bzheng888** [comment]
Hi， @jiacheliu3， thanks for your reply, I'm not very clear what your concerns. If we use ASYNC_THROUGH or MUST_CACHE, and no matter ufs is readonly, it has a risk to lead data loss any way. For example, if the ufs is not readonly, and we use ASYNC_THROUGH write, then we can remove data to lead a data loss, and why when the ufs is readonly, we can't remove the in alluxio data?

**Comment by HelloHorizon** [comment]
@jiacheliu3 can you take another look?

**Comment by jiacheliu3** [comment]
Currently blocked by https://github.com/Alluxio/alluxio/issues/14992 Need a better definition of "readonly" before we can determine the correct behavior in this case. This issue needs to be fixed. We are only not sure how.

File: core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java
Comment (jiacheliu3) [inline core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java:1795]: Pls leave a comment here
```
// If the mount point is read only, we allow removing the in-Alluxio metadata and data
// in order to load it from the UFS again.
// This can happen if Alluxio is out-of-sync with the UFS.
```

**Review by jiacheliu3 — APPROVED** [review]
LGTM with a minor comment, thanks for the fix!

**Comment by HelloHorizon** [comment]
@bzheng888 can you fix the conflicting files?

**Comment by bzheng888** [comment]
@jiacheliu3 PTAL, thanks!

**Review by jiacheliu3 — APPROVED** [review]
LGTM, will merge when tests all pass

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**
